context/src/test/java/io/grpc/PersistentHashArrayMappedTrieTest.java

```diff
@@ -84,15 +84,23 @@ public void leaf_insert() {
     assertEquals(2, ret.size());
   }
 
-  @Test(expected = AssertionError.class)
+  @Test
   public void collisionLeaf_assertKeysDifferent() {
     Key key1 = new Key(0);
-    new CollisionLeaf<>(key1, new Object(), key1, new Object());
+    try {
+      new CollisionLeaf<>(key1, new Object(), key1, new Object());
+      throw new Error();
```

Comment:
I usually do something like
```suggestion
      fail("Expected AssertionError");
```

Reply:
Actually, already got bitten by that. `fail()` throws an AssertionError so it would be thrown away. The workaround is to manually throw some other throwable instead (in this case, Error)

Reply:
Nice! TIL.